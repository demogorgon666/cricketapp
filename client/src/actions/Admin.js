import {
  GET_PLAYERS,
  CREATE_PLAYER,
  CREATE_TEAM,
  EDIT_PLAYER,
  EDIT_TEAM,
  DELETE_PLAYER,
  DELETE_TEAM,
  GET_PLAYER_SEARCH,
  GET_TEAM_SEARCH,
  GET_ALLTEAMS,
  ERROR_TYPE
} from "./Types";
import axios from "axios";

// export const getPlayers = () => dispatch => {
// 	return axios
// 		.get("http://localhost:5000/api/cricketalpha/allPlayer")
// 		.then(res => {
// 			console.log("From Player Actions", res.data.data);
// 			dispatch({
// 				type: GET_PLAYERS,
// 				payload: res.data.data
// 			});
// 		})
// 		.catch(err => {
// 			console.log(err);
// 		});
// };
//
export const getPlayers = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

// export const getTeams = () => dispatch => {
//     return axios
//       .get("http://localhost:5000/api/cricketalpha/allteams")
//       .then(res => {
//         dispatch({
//           type: GET_TEAMS,
//           payload: res.data.data
//         });
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

export const getAllTeams = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/admin/allteam", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_ALLTEAMS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

//   export const createPlayer = player => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/player/new", player)
//       .then(res => {
//         dispatch({
//           type: CREATE_PLAYER
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

//   export const createTeam = team => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/user/new", team)
//       .then(res => {
//         dispatch({
//           type: CREATE_TEAM
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };
export const createTeam = team => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/team/new", team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_TEAM
      });
      alert("Added one team successfully");
      dispatch(getAllTeams());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const createPlayer = player => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/player/new", player, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_PLAYER
      });
      alert("Added one player successfully");
      dispatch(getPlayers());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const editPlayer = player => dispatch => {
  return axios
    .put(
      "http://localhost:5000/apis/admin/editplayer/" + player.player_id,
      player,
      {
        headers: {
          Authorization: `Bearer ${localStorage.getItem("token")}`
        }
      }
    )
    .then(res => {
      dispatch({
        type: EDIT_PLAYER
      });

      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const editTeam = team => dispatch => {
  return axios
    .put("http://localhost:5000/apis/admin/editteam/" + team.team_id, team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: EDIT_TEAM
      });
      // dispatch(getUsers());
      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deletePlayer = player_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteplayer/" + player_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_PLAYER
      });
      dispatch(getPlayers());
      // alert("Player Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deleteTeam = team_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteteam/" + team_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_TEAM
      });
      dispatch(getAllTeams());
      // alert("Team Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const getPlayerSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/search/" + searchString)
    .then(res => {
      dispatch({
        type: GET_PLAYER_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTeamSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/searchteam/" + searchString)
    .then(res => {
      dispatch({
        type: GET_TEAM_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
