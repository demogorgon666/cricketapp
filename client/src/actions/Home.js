import {
  GET_PLAYERS,
  GET_STATISTICS,
  GET_PLAYER_STATS,
  GET_TOP_BATSMAN,
  GET_NEWS,
  GET_HOME_RECENT_MATCHES
} from './Types'
import axios from 'axios'

export const getNews = () => dispatch => {
  axios
    .get('http://localhost:5000/apis/news')
    .then(res => {
      dispatch({
        type: GET_NEWS,
        payload: res.data.data
      })
      console.log("from 'then' get news")
    })

    .catch(err => console.log(err))
}

// fetching recent matches
export const getRecentMatches = () => dispatch => {
  console.log('from actions of home recent matches')
  axios
    .get('http://localhost:5000/apis/recentMatches')
    .then(res => {
      dispatch({
        type: GET_HOME_RECENT_MATCHES,
        payload: res.data.data
      })
      console.log("from 'then' get RECENT MATCHES")
    })

    .catch(err => console.log(err))
}
