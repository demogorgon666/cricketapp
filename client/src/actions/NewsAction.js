import { GET_NEWS_ID } from "./Types";
import axios from "axios";
import { id } from "date-fns/esm/locale";

export const getNewsById = id => dispatch => {
  console.log("getnews by id", id);
  return axios
    .get("http://localhost:5000/apis/News/newsbyid/" + id)
    .then(res => {
      console.log("From newsbyid actions", res.data.data);
      dispatch({
        type: GET_NEWS_ID,
        payload: res.data.data,
      });
    })
    .catch(err => {
      console.log(err);
    });
};
