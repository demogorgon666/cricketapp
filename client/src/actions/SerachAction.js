import { GET_SEARCH } from "./Types";
import axios from "axios";
import Search from "../reducers/Search";

export const getSearch = (search_term, history) => dispatch => {
	console.log("actions getsearch", search_term);
	return axios
		.post("http://localhost:5000/apis/Search/search", search_term)
		.then(res => {
			console.log("From Search Actions", res.data.data);
			dispatch({
				type: GET_SEARCH,
				payload: res.data.data
			});
			console.log("actions history", history);
			console.log("res", res.data.data[0].player_id);

			history.push("/playerInfo/" + res.data.data[0].player_id);
		})
		.catch(err => {
			console.log(err);
		});
};
