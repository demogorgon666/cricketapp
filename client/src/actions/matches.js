import { GET_RECENT_MATCHES, GET_MATCHES_DATE, GET_MATCH_DETAILS_BY_ID } from './Types'
import axios from "axios";

const url = 'http://localhost:5000/api/matches';

export const getRecentMatches = (date) => dispatch => {
    return axios
        .get(url + "/recent/" + date, {
        })
        .then(res => {
            dispatch({
                type: GET_RECENT_MATCHES,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getMatchesDate = () => dispatch => {
    return axios
        .get(url + "/bydate", {
        })
        .then(res => {
            dispatch({
                type: GET_MATCHES_DATE,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getmatchdetailbyId = (id) => dispatch => {
    return axios
        .get(url + "/summary/" + id, {
        })
        .then(res => {
            dispatch({
                type: GET_MATCH_DETAILS_BY_ID,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

