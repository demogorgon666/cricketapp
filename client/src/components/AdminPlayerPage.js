import React, { Component } from "react";
// import { Link } from "react-router-dom";
import {
  getPlayers,
  createPlayer,
  editPlayer,
  deletePlayer,
  getPlayerSearch
} from "../actions/Admin";
import { connect } from "react-redux";
import "./css/Adminpage.css";
// import NavBar from "./common/Navbar";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import SweetAlert from "sweetalert-react";
import "sweetalert/dist/sweetalert.css";
import Loader from "react-loader-spinner";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

export class AdminPlayerPage extends Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);

    this.state = {
      player_gender: "Gender",
      batting_style: "Batting Style",
      bowling_style: "Bowling style",
      player_role: "Player Role",
      searchString: ""
    };
    this.OnSelectGender = this.OnSelectGender.bind(this);
    this.OnSelectBattingStyle = this.OnSelectBattingStyle.bind(this);
    this.OnSelectBowlerStyle = this.OnSelectBowlerStyle.bind(this);
    this.OnSelectRole = this.OnSelectRole.bind(this);
    this.OnChange = this.OnChange.bind(this);
  }
  OnSelectGender(e) {
    this.setState({
      player_gender: e.target.value
    });
  }
  OnSelectBattingStyle(e) {
    this.setState({
      batting_style: e.target.value
    });
  }
  OnSelectBowlerStyle(e) {
    this.setState({
      bowling_style: e.target.value
    });
  }
  OnSelectRole(e) {
    this.setState({
      player_role: e.target.value
    });
  }

  componentDidMount() {
    if (!localStorage.getItem("token")) {
      this.props.history.push("/login");
    }
    this.props.getPlayers();
  }

  state = {
    player_name: "",
    player_country: "",
    batting_style: "",
    bowling_style: "",
    player_gender: "",
    player_role: "",
    debut_odi_match: "",
    debut_test_match: "",
    debut_t20_match: "",
    showError: false,
    errorMessage: "",
    player_dob: new Date(),
    searchString: "",
    show: false
  };

  OnChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleChange = date => {
    this.setState({
      player_dob: date
    });
  };

  onRegister = e => {
    e.preventDefault();
    if (
      !this.state.player_name ||
      !this.state.player_country ||
      !this.state.player_gender ||
      !this.state.batting_style ||
      !this.state.bowling_style ||
      !this.state.player_dob ||
      !this.state.player_role ||
      !this.state.debut_odi_match ||
      !this.state.debut_test_match ||
      !this.state.debut_t20_match
    ) {
      this.setState({
        showError: true,
        errorMessage: "Enter all the fields"
      });
    } else {
      //   e.preventDefault();

      // e.preventDefault();
      if (this.state.player_name.length < 2) {
        this.setState({
          showError: true,
          errorMessage: "Name length should be greater then 1!"
        });
      } else {
        if (this.state.player_gender.length < 2) {
          this.setState({
            showError: true,
            errorMessage: "Please enter valid player type!"
          });
        } else {
          if (this.state.player_country.length < 2) {
            this.setState({
              showError: true,
              errorMessage: "Please enter valid player type!"
            });
          } else {
            if (this.state.batting_style.length < 5) {
              this.setState({
                showError: true,
                errorMessage: "Please enter valid batting style!"
              });
            } else {
              if (this.state.bowling_style.length < 5) {
                this.setState({
                  showError: true,
                  errorMessage: "Please enter valid bowling style!"
                });
              } else {
                if (this.state.player_dob.length < 5) {
                  this.setState({
                    showError: true,
                    errorMessage: "Please enter valid player dob!"
                  });
                } else {
                  if (this.state.player_role.length < 5) {
                    this.setState({
                      showError: true,
                      errorMessage: "Please enter valid player role!"
                    });
                  } else {
                    if (this.state.debut_odi_match.length < 5) {
                      this.setState({
                        showError: true,
                        errorMessage: "Please enter valid debut odi match!"
                      });
                    } else {
                      if (this.state.debut_test_match.length < 5) {
                        this.setState({
                          showError: true,
                          errorMessage: "Please enter valid debut test match!"
                        });
                      } else {
                        if (this.state.debut_t20_match.length < 5) {
                          this.setState({
                            showError: true,
                            errorMessage: "Please enter valid debut t20 match!"
                          });
                        } else {
                          this.setState({
                            showError: false,
                            errorMessage: ""
                          });
                          let player = {
                            player_name: this.state.player_name,
                            player_country: this.state.player_country,
                            batting_style: this.state.batting_style,
                            bowling_style: this.state.bowling_style,
                            player_dob: this.state.player_dob.toLocaleDateString(
                              "en-GB"
                            ),
                            player_role: this.state.player_role,
                            player_gender: this.state.player_gender,
                            debut_odi_match: this.state.debut_odi_match,
                            debut_test_match: this.state.debut_test_match,
                            debut_t20_match: this.state.debut_t20_match
                          };
                          this.props.createPlayer(player);
                          //   this.props.history.push("/login");
                          this.setState({
                            player_name: "",
                            player_country: "",
                            batting_style: "",
                            bowling_style: "",
                            player_dob: "",
                            player_gender: "",
                            player_role: "",
                            debut_odi_match: "",
                            debut_test_match: "",
                            debut_t20_match: ""
                          });
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  };

  onSearchInputChange = event => {
    if (event.target.value) {
      this.setState({ searchString: event.target.value });
      this.props.getPlayerSearch(this.state.searchString);
    } else {
      this.setState({ searchString: "" });
    }
  };

  render() {
    console.log("loading is:", this.props.isLoading);
    return (
      <div>
        {/* <NavBar /> */}
        <div className="player-page">
          {this.props.isLoading ? (
            <div style={{ margin: "auto" }}>
              <Loader
                type="TailSpin"
                color="#2980b9"
                height="100"
                width="100"
              />
            </div>
          ) : (
            <div className="players">
              <div className="inner-heading">
                <div>
                  <h1 id="heading-1">All Players</h1>
                </div>
                <div>
                  {" "}
                  <input
                    id="searchInput"
                    type="text"
                    name="Search"
                    margin="normal"
                    placeholder="Search"
                    onChange={this.onSearchInputChange}
                  />
                </div>
              </div>

              <div className="player-list">
                {this.props.playerInfo.map(players => (
                  <div className="player-name">
                    <p>{players.player_name}</p>
                    <div className="inner-button">
                      <div style={{ marginRight: "5px" }}>
                        <button
                          id="editbutton"
                          className="admininnerbutton"
                          onChange={this.OnChange}
                          onClick={() => {
                            this.props.history.push(
                              "/admineditplayer/" + players.player_id,
                              { players }
                            );
                          }}
                        >
                          Edit
                        </button>
                        {/* <Link
                        className="link-button"
                        to={{
                          pathname: "/admineditplayer/" + players.player_id,
                          state: { players }
                        }}
                      >
                        Edit
                      </Link> */}
                      </div>

                      <div>
                        <button
                          id="deletebutton"
                          className="admininnerbutton"
                          onChange={this.OnChange}
                          // onClick={() =>
                          //   this.props.deletePlayer(players.player_id)
                          // }
                          onClick={() => this.setState({ show: true })}
                          style={{ background: "#E74C3c" }}
                        >
                          Delete
                        </button>
                        <SweetAlert
                          id="SweetAlert"
                          show={this.state.show}
                          type="warning"
                          title={`DELETE PLAYER!`}
                          text="Are you sure?"
                          confirmButtonText="Delete!"
                          confirmButtonColor="#e74c3c"
                          cancelButtonText="No, keep it"
                          showCancelButton
                          onConfirm={() => {
                            // console.log("confirm");
                            this.setState({ show: false });
                            this.props.deletePlayer(players.player_id);
                          }}
                          onCancel={() => {
                            // console.log("cancel");

                            this.setState({ show: false });
                          }}
                          onEscapeKey={() => this.setState({ show: false })}
                          onOutsideClick={() => this.setState({ show: false })}
                        />
                      </div>
                    </div>
                  </div>
                ))}
              </div>
            </div>
          )}

          <div id="Adminform">
            <fieldset>
              <h2 id="heading-2">Add New Player</h2>
              <input
                type="text"
                name="player_name"
                id="enter-name"
                placeholder="Enter Name"
                value={this.state.player_name}
                onChange={this.OnChange}
              />
              <input
                type="text"
                id="player_country"
                name="player_country"
                placeholder="Enter Player Country"
                value={this.state.player_country}
                onChange={this.OnChange}
              />
              <div className="playing-style">
                <div>
                  <select
                    type="text"
                    id="batting_style"
                    name="batting_style"
                    value={this.state.batting_style}
                    onChange={this.OnSelectBattingStyle}
                    className="choice"
                  >
                    <option name="choice">Batting style</option>
                    <option name="male">Right Hand</option>
                    <option name="male">left Hand</option>
                  </select>
                </div>
                <div>
                  <select
                    id="bowling_style"
                    type="text"
                    name="bowling_style"
                    value={this.state.bowling_style}
                    onChange={this.OnSelectBowlerStyle}
                    className="choice"
                  >
                    <option name="choice">Bowling style</option>
                    <option name="male">Right-arm fast</option>
                    <option name="male">Right-arm fast medium</option>
                    <option name="male">Right-arm medium fast</option>
                    <option name="male">Right-arm medium</option>
                    <option name="male">Left-arm fast</option>
                    <option name="male">Left-arm fast medium</option>
                    <option name="male">Left-arm medium fast</option>
                    <option name="male">Left-arm medium</option>
                    <option name="male">Off break (right-arm)</option>
                    <option name="male">Leg break (right-arm)</option>
                    <option name="male">Slow left-arm orthodox</option>
                    <option name="male">Slow left-arm chinaman</option>
                  </select>
                </div>
              </div>
              <div className="playing-style">
                <DatePicker
                  id="DatePicker"
                  name="datepicker"
                  dateFormat="yyyy-MM-dd"
                  placeholderText="DOB"
                  selected={this.state.player_dob}
                  onChange={this.handleChange}
                />

                <div>
                  <select
                    id="player_gender"
                    type="text"
                    name="player_gender"
                    value={this.state.player_gender}
                    onChange={this.OnSelectGender}
                    className="choice"
                  >
                    <option name="choice">Gender</option>
                    <option name="male">male</option>
                    <option name="male">female</option>
                  </select>
                </div>
              </div>
              <select
                id="player_role"
                type="text"
                name="player_role"
                placeholder="Enter Player Role"
                value={this.state.player_role}
                onChange={this.OnSelectRole}
                className="choice"
              >
                <option name="choice">Player Role</option>
                <option name="male">Wicket-keeper-batsmen</option>
                <option name="male">Wicket-keeper-batswomen</option>
                <option name="male">Bowler</option>
                <option name="male">Batsmen</option>
                <option name="male">Batswomen</option>
                <option name="male">Allrounder</option>
              </select>
              <input
                id="debut_odi_match"
                type="text"
                name="debut_odi_match"
                placeholder="Enter Debut ODI Match"
                value={this.state.debut_odi_match}
                onChange={this.OnChange}
              />
              <input
                id="debut_test_match"
                type="text"
                name="debut_test_match"
                placeholder="Enter Debut Test Match"
                value={this.state.debut_test_match}
                onChange={this.OnChange}
              />
              <input
                id="debut_t20_match"
                type="text"
                name="debut_t20_match"
                placeholder="Enter Debut T20 Match"
                value={this.state.debut_t20_match}
                onChange={this.OnChange}
              />
              <div
                className="errorMessage"
                style={{
                  color: "#c0392b"
                }}
              >
                {/*dispatch error from node*/}
                {this.props.error ? (
                  <>{this.props.error}</>
                ) : (
                  <span
                    className="errorMessage"
                    style={{
                      color: "#c0392b",
                      display: this.state.showError ? "block" : "none"
                    }}
                  >
                    {this.state.errorMessage}
                  </span>
                )}
              </div>
              <button
                id="addbutton"
                className="adminpagebutton "
                onChange={this.OnChange}
                onClick={this.onRegister}
              >
                Add Player
                {/* {console.log(this.onRegister)} */}
              </button>
              {/* <p>
              Already have an account ?
              <Link to="/login">
                <span
                  style={{
                    color: "#2980b9",
                    fontWeight: "bold"
                  }}
                >
                  Login
                </span>
              </Link>
            </p> */}
            </fieldset>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  player: state.AdminPlayerReducer.player,
  playerInfo: state.PlayerReducer.playerInfo,
  error: state.AdminPlayerReducer.error,
  isLoading: state.LoadingReducer.isLoading
});

export default connect(
  mapStateToProps,
  { getPlayers, createPlayer, editPlayer, deletePlayer, getPlayerSearch }
)(AdminPlayerPage);
