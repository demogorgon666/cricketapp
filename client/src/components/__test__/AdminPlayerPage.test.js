import React from "react";
import { shallow } from "enzyme";
import { AdminPlayerPage } from "../AdminPlayerPage";
// import store from "../../store";
// import { Provider } from "react-redux";
import { getPlayers } from "../../actions/Admin";
const adminplayerpage = jest.fn();
let playerInfo = [];

const wrapper = shallow(
  <AdminPlayerPage
    adminplayerpage={adminplayerpage}
    playerInfo={playerInfo}
    getPlayers={getPlayers}
  />
);

describe("test  Component", () => {
  it("render the component", () => {
    // const wrapper = shallow(<Register register={register} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("there should be a div", () => {
    expect(wrapper.find("div").length).toBe(13);
  });
  // it("there should be a navbar", () => {
  //   expect(wrapper.find("NavBar").length).toBe(1);
  // });
  it("there should be a h1", () => {
    expect(wrapper.find("h1").length).toBe(2);
    expect(
      wrapper
        .find("h1")
        .at(0)
        .text()
    ).toBe("All Players");
    expect(
      wrapper
        .find("h1")
        .at(1)
        .text()
    ).toBe("Add New Player!");
  });
  it("there should be a input", () => {
    expect(wrapper.find("input").length).toBe(6);
  });
  it("there should be a buttons", () => {
    expect(wrapper.find("button").length).toBe(1);
  });
  // it("there should be a link", () => {
  //   expect(wrapper.find("Link").length).toBe(2);
  // });

  it("there should be a fieldset tag", () => {
    expect(wrapper.find("fieldset").length).toBe(1);
  });
  it("there should be a select tag", () => {
    expect(wrapper.find("select").length).toBe(4);
  });
  // it("there should be a p tag", () => {
  //   expect(wrapper.find("p").length).toBe(1);
  //   expect(wrapper.find("p").text()).toBe("Already have an account ?Login");
  // });
  it("there should be a button", () => {
    expect(wrapper.find("button").text()).toBe("Add Player");
  });

  // it("should change the state of testClicked to true when test tab is clicked", () => {
  //   const test = wrapper.find("button").at(0);
  //   test.simulate("click");
  //   expect(wrapper.state().onRegister).toBe(true);
  // });

  it("there should be a span", () => {
    expect(wrapper.find("span").length).toBe(1);
  });

  it("there should type inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(2)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(3)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(4)
        .prop("type")
    ).toEqual("text");
  });

  it("there should name in  inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("name")
    ).toEqual("Search");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("name")
    ).toEqual("player_name");
    expect(
      wrapper
        .find("input")
        .at(2)
        .prop("name")
    ).toEqual("player_country");

    expect(
      wrapper
        .find("input")
        .at(3)
        .prop("name")
    ).toEqual("debut_odi_match");
    expect(
      wrapper
        .find("input")
        .at(4)
        .prop("name")
    ).toEqual("debut_test_match");
    expect(
      wrapper
        .find("input")
        .at(5)
        .prop("name")
    ).toEqual("debut_t20_match");
  });
  it("there should name in  inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("placeholder")
    ).toEqual("Search");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("placeholder")
    ).toEqual("Enter Name");
    expect(
      wrapper
        .find("input")
        .at(2)
        .prop("placeholder")
    ).toEqual("Enter Player Country");

    expect(
      wrapper
        .find("input")
        .at(3)
        .prop("placeholder")
    ).toEqual("Enter Debut ODI Match");
    expect(
      wrapper
        .find("input")
        .at(4)
        .prop("placeholder")
    ).toEqual("Enter Debut Test Match");
    expect(
      wrapper
        .find("input")
        .at(5)
        .prop("placeholder")
    ).toEqual("Enter Debut T20 Match");
  });
});
