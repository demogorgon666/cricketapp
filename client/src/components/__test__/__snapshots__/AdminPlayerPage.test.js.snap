// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  Component render the component 1`] = `
<div>
  <Navbar />
  <div
    className="player-page"
  >
    <div
      className="players"
    >
      <div
        className="inner-heading"
      >
        <div>
          <h1>
            All Players
          </h1>
        </div>
        <div>
           
          <input
            margin="normal"
            name="Search"
            onChange={[Function]}
            placeholder="Search"
            type="text"
          />
        </div>
      </div>
      <div
        className="player-list"
      />
    </div>
    <div
      id="Adminform"
    >
      <fieldset>
        <h1>
          Add New Player!
        </h1>
        <input
          name="player_name"
          onChange={[Function]}
          placeholder="Enter Name"
          type="text"
        />
        <input
          name="player_country"
          onChange={[Function]}
          placeholder="Enter Player Country"
          type="text"
        />
        <div
          className="playing-style"
        >
          <div>
            <select
              className="choice"
              name="batting_style"
              onChange={[Function]}
              type="text"
              value="Batting Style"
            >
              <option
                name="choice"
              >
                Batting style
              </option>
              <option
                name="male"
              >
                Right Hand
              </option>
              <option
                name="male"
              >
                left Hand
              </option>
            </select>
          </div>
          <div>
            <select
              className="choice"
              name="bowling_style"
              onChange={[Function]}
              type="text"
              value="Bowling style"
            >
              <option
                name="choice"
              >
                Bowling style
              </option>
              <option
                name="male"
              >
                Right-arm fast
              </option>
              <option
                name="male"
              >
                Right-arm fast medium
              </option>
              <option
                name="male"
              >
                Right-arm medium fast
              </option>
              <option
                name="male"
              >
                Right-arm medium
              </option>
              <option
                name="male"
              >
                Left-arm fast
              </option>
              <option
                name="male"
              >
                Left-arm fast medium
              </option>
              <option
                name="male"
              >
                Left-arm medium fast
              </option>
              <option
                name="male"
              >
                Left-arm medium
              </option>
              <option
                name="male"
              >
                Off break (right-arm)
              </option>
              <option
                name="male"
              >
                Leg break (right-arm)
              </option>
              <option
                name="male"
              >
                Slow left-arm orthodox
              </option>
              <option
                name="male"
              >
                Slow left-arm chinaman
              </option>
            </select>
          </div>
        </div>
        <div
          className="playing-style"
        >
          <t
            allowSameDay={false}
            dateFormat="yyyy-MM-dd"
            dateFormatCalendar="LLLL yyyy"
            disabled={false}
            disabledKeyboardNavigation={false}
            dropdownMode="scroll"
            inlineFocusSelectedMonth={false}
            monthsShown={1}
            name="datepicker"
            nextMonthButtonLabel="Next Month"
            nextYearButtonLabel="Next Year"
            onBlur={[Function]}
            onChange={[Function]}
            onClickOutside={[Function]}
            onFocus={[Function]}
            onInputClick={[Function]}
            onInputError={[Function]}
            onKeyDown={[Function]}
            onMonthChange={[Function]}
            onSelect={[Function]}
            onYearChange={[Function]}
            placeholderText="DOB"
            preventOpenOnFocus={false}
            previousMonthButtonLabel="Previous Month"
            previousYearButtonLabel="Previous Year"
            readOnly={false}
            renderDayContents={[Function]}
            shouldCloseOnSelect={true}
            showMonthYearPicker={false}
            showQuarterYearPicker={false}
            showTimeInput={false}
            showTimeSelect={false}
            strictParsing={false}
            timeCaption="Time"
            timeInputLabel="Time"
            timeIntervals={30}
            withPortal={false}
          />
          <div>
            <select
              className="choice"
              name="player_gender"
              onChange={[Function]}
              type="text"
              value="Gender"
            >
              <option
                name="choice"
              >
                Gender
              </option>
              <option
                name="male"
              >
                male
              </option>
              <option
                name="male"
              >
                female
              </option>
            </select>
          </div>
        </div>
        <select
          className="choice"
          name="player_role"
          onChange={[Function]}
          placeholder="Enter Player Role"
          type="text"
          value="Player Role"
        >
          <option
            name="choice"
          >
            Player Role
          </option>
          <option
            name="male"
          >
            Wicket-keeper-batsmen
          </option>
          <option
            name="male"
          >
            Wicket-keeper-batswomen
          </option>
          <option
            name="male"
          >
            Bowler
          </option>
          <option
            name="male"
          >
            Batsmen
          </option>
          <option
            name="male"
          >
            Batswomen
          </option>
          <option
            name="male"
          >
            Allrounder
          </option>
        </select>
        <input
          name="debut_odi_match"
          onChange={[Function]}
          placeholder="Enter Debut ODI Match"
          type="text"
        />
        <input
          name="debut_test_match"
          onChange={[Function]}
          placeholder="Enter Debut Test Match"
          type="text"
        />
        <input
          name="debut_t20_match"
          onChange={[Function]}
          placeholder="Enter Debut T20 Match"
          type="text"
        />
        <span
          className="errorMessage"
          style={
            Object {
              "color": "#c0392b",
              "display": "none",
            }
          }
        />
        <button
          onChange={[Function]}
          onClick={[Function]}
        >
          Add Player
        </button>
      </fieldset>
    </div>
  </div>
</div>
`;
